### ISSUE TITLE
Resource caching not clearing on changes to pipeline resource_type

### ISSUE THREAD

**Issue by devdavidkarlsson** [concourse/concourse#913]
# Bug Report

I have had some issues when trying to POC stuff in concourse and the resource_type has been misconfigured. It appears that it becomes cached in concourse, doing a set pipeline with changes to the same resource/resource-type does not seem to flush the cache... This can result in pending resources...

**Example:**

```
resource_types:
- name: nm_cache
  type: docker-image
  source: 
    repository: {{docker-registry-npm-cache-non-working}}
    insecure_registries:
       - {{my-docker-registry}}
```

update the pipeline to the working repository, set pipeline using fly:

```
resource_types:
- name: nm_cache
  type: docker-image
  source: {repository: devdavid/npm-cache-resource, tag: latest}
```

**Result**
Pipeline is still pending and cannot start even though cancelled, retriggered manually....

**Expected:**
The new resource type is used by the resources utilizing the resource_type named: nm_cache (in this case)...

* Concourse version: 2.60

**Comment by devdavidkarlsson** [comment]
Hi guys how can i Pop a version of a resource from the cache?
I want it gone..
just removing the pipeline and adding it again does not remove it's cached resources from the cache 🤕

**Comment by ClaraFuPivotal** [comment]
I think this was fixed by https://github.com/concourse/atc/commit/a1498508183d1b3e8d64d4fde1135fd2ef8489ae. We happened to find this while investigating something else so the fix will end up in 2.8.
